taf/updater/handlers.py

```diff
@@ -68,82 +84,170 @@ def __init__(self, mirrors, repository_directory):
     self.current_path = os.path.join(metadata_path, 'current')
     self.previous_path = os.path.join(metadata_path, 'previous')
     os.mkdir(self.current_path)
     os.mkdir(self.previous_path)
 
-    for filename in glob.glob(os.path.join(metadata_path, '*.json')):
-      shutil.copy(filename, self.current_path)
-      shutil.copy(filename, self.previous_path)
+    metadata_files = self.validation_auth_repo.list_files_at_revision(self.current_commit, 'metadata')
+    for filename in metadata_files:
+      metadata = self.validation_auth_repo.get_file(self.current_commit, f'metadata/{filename}')
+      current_filename = os.path.join(self.current_path, filename)
+      previous_filename = os.path.join(self.previous_path, filename)
+      with open(current_filename, 'w') as f:
+        f.write(metadata)
+      shutil.copyfile(current_filename, previous_filename)
 
-    self.users_auth_repo = GitRepo(repository_directory)
-    self.users_auth_repo.is_git_repository()
-    self.users_auth_repo.checkout_branch('master')
-    self._init_commits()
 
   def _clone_validation_repo(self, url):
+    """
+    Clones the authentication repository based on the url specified using the
+    mirrors parameter. The repository is cloned as a bare repository
+    to a the temp directory and will be deleted one the update is done.
+    """
     temp_dir = tempfile.gettempdir()
-    self.validation_auth_repo = BareGitRepo(temp_dir)
-    self.validation_auth_repo.clone(url)
+    repo_name = self.users_auth_repo.name
+    self.validation_auth_repo = AuthenticationRepo(temp_dir, self.metadata_path, self.targets_path,
+                                                   repo_name, [url], True)
+    self.validation_auth_repo.clone()
     self.validation_auth_repo.fetch(fetch_all=True)
 
-  def _init_commits(self):
-    # TODO handle the case when the local repository does not
-    # exist and needs to be cloned
-    # for now, it is assumed that there is a local repository
-    users_head_sha = self.users_auth_repo.head_commit_sha()
-    # find all commits after the top commit of the
-    # client's local authentication repository
-    # TODO detect force-push removal of commits
-    self.commits = self.validation_auth_repo. \
-      all_commits_since_commit(users_head_sha)
-      # insert the current one at the beginning of the list
-    self.commits.insert(0, users_head_sha)
-    self.commits_indexes = {}
-    for file_name in self.users_auth_repo. \
-        list_files_at_revision(users_head_sha):
-      self.commits_indexes[file_name] = 0
-
-  def get_mirrors(self, remote_filename):
+  def cleanup(self):
+    """
+    Removes the bare authentication repositoy and current and previous
+    directories. This should be called after the update is finished,
+    either successfully or unsuccessfully.
+    """
+    shutil.rmtree(self.current_path)
+    shutil.rmtree(self.previous_path)
+    os.system(f'rmdir /S /Q "{self.validation_auth_repo.repo_path}"')
```

Comment:
Why you didn't use `shutil.rmtree` here as well?

Reply:
It didn't work. Was getting a permission error

Reply:
What I can see from code is that you are using `tempfile` to create a temporary directory [here](https://github.com/openlawlibrary/taf/blob/renatav/updater/taf/updater/handlers.py#L185). 

```python
>>> import tempfile
>>> print(tempfile.gettempdir())
/var/folders/lc/c_lvyn5x6cbd12ysjly61qcr0000gn/T
```

This is what I get on my machine (macOS).
I suggest adding another child directory, with a random name, like: `/var/folders/lc/c_lvyn5x6cbd12ysjly61qcr0000gn/T/awled12k34n1`, to avoid possible conflicts.

> It didn't work. Was getting a permission error

Can you try this:

```python
temp_dir = tempfile.mkdtemp() # does what I mentioned above; instead of tempfile.gettempdir()
self.validation_auth_repo = AuthenticationRepo(temp_dir, self.metadata_path, self.targets_path, repo_name, [url], True)

shutil.rmtree(self.validation_auth_repo.repo_path)
```

@renatav I will push it on [my branch](https://github.com/openlawlibrary/taf/blob/danixeee/setup-formatter-and-bandit) where I fixed some `pylint` and `bandit` issues for the updater as well. Please test when you have time.

Reply:
I liked cloning the repository directly inside temp, as that made testing easier (it's easier to find it inside temp, see what it looks like, delete it manually if needed etc.). Update of the authentication repository seems stable now, so I will look into fixing this. I'll test if your method works on Windows